File: my-solutions/0121-best-time-to-buy-and-sell-stock/solution.java
public class Solution {
    public int maxProfit(int[] prices) {
        
        if(prices.length < 2)
            return 0;
            
        int profit = 0;
        int minElement = Integer.MAX_VALUE;
    for(int i=0; i<prices.length; i++){
       profit = Math.max(profit, prices[i]-minElement);
       minElement = Math.min(minElement, prices[i]);
    }
        return profit;
    }
}

File: my-solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        # Need to keep track of min price and max profit we can get
        min_price = math.inf
        max_profit = 0

        for price in prices:

            min_price = min(min_price, price)
            max_profit = max(max_profit, price - min_price)

        return max_profit

        
